### DISCUSSION TITLE
The API docs mentions "This key is safe to use in a browser if you have enabled Row Level Security"

### DISCUSSION THREAD

**Discussion by santoshyadavdev** [supabase/supabase#1507] [Questions] [answered]
I am trying to understand this, the doc says "This key is safe to use in a browser if you have enabled Row Level Security" but if I enable the Row Level Security I am not even able to select the data.

What's the purpose of Row Level Security, can someone explain to me in detail, I am calling API from JS and users can access the API key from the request sent, what someone tries to make the API call with the endPoint and API key.

Or is there anything I am missing?

**Comment by easy-street** [comment, marked as answer]
When RLS is enabled, table access must be allowed by a policy. From the [PostgreSQL Row Security Policy docs](https://www.postgresql.org/docs/12/ddl-rowsecurity.html):
> If no policy exists for the table, a default-deny policy is used, meaning that no rows are visible or can be modified

To learn more about RLS and creating RLS policies with Supabase, definitely check out these resources:
- https://supabase.io/docs/guides/auth#row-level-security
- https://supabase.io/docs/learn/auth-deep-dive/auth-row-level-security
- https://supabase.io/docs/learn/auth-deep-dive/auth-policies

**Reply by santoshyadavdev** [reply]
Thanks a lot 😊
